{"name": "test_delete_all_pets_positive", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'", "trace": "self = <tests.test_my_pets_positive.TestMyPetsPagePositive object at 0x0000020555E4C5E0>, driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"f2b85e6c7f4b03cfca73945045829390\")>\n\n    @pytest.mark.four\n    @pytest.mark.delete_all_pets\n    @allure.feature('Удаление карточек питомцев_POSITIVE TESTS')\n    @allure.story(\"Удаление всех карточек питомцев из профиля пользователя (очистка профиля)\")\n    @allure.severity(allure.severity_level.NORMAL)\n    def test_delete_all_pets_positive(self, driver):\n        \"\"\"Позитивный тест проверки удаления пользователем всех созданных им карточек питомцев. Валидация теста\n        выполнена успешно в случае, если после последовательного воздействия на элемент \"Удалить питомца\" в каждой\n        карточке питомца, все карточки будут удалены из стека пользователя. Тест предусматривает проверку количества\n        карточек до и после удаления.\n        P.S. По большей части тест создан для очистки профиля пользователя от ранее созданных карточек.\"\"\"\n    \n        with allure.step(\"Шаг 1: Проверка количества карточек питомцев в профиле (если нет ни одной -> Exception).\"):\n            page = MyPetsPage(driver)\n            page.wait_page_loaded()\n            cards_before_delete = page.get_pets_quantity(driver)\n            pets_quantity = page.get_pets_quantity(driver)\n            if pets_quantity == 0:\n                screenshot = page.make_screenshot()\n                allure.attach(screenshot,\n                              name=\"delete_all_pets_FAILED\",\n                              attachment_type=allure.attachment_type.PNG)\n                raise Exception(\"Добавленные(ый) пользователем питомцы(ец) отсутствуют(ет), нет ни одной карточки для \"\n                                \"удаления!\")\n        with allure.step(\"Шаг 2: Удаление всех карточек питомцев из профиля\"):\n            while pets_quantity != 0:\n                page.delete_pet_btn_click(driver)\n                page.refresh_page()\n                page.wait_page_loaded(wait_for_element=page.add_pet_btn)\n                pets_quantity = page.get_pets_quantity(driver)\n            cards_after_delete = page.get_pets_quantity(driver)\n        with allure.step(\"Шаг 3: Assert-проверка количества карточек до/после удаления.\"):\n            assert cards_after_delete == 0, \"Ошибка, не все карточки удалены!\"\n            screenshot = page.make_screenshot()\n>           allure.attach(screenshot,\n                          name=\"delete_all_pets_PASSED\",\n                          attachment_type=allure.attachment_type.PNG)\n\ntest_my_pets_positive.py:197: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\_allure.py:200: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_hooks.py:493: in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_manager.py:115: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\allure_pytest\\listener.py:249: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\reporter.py:165: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_hooks.py:493: in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_manager.py:115: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <allure_commons.logger.AllureFileLogger object at 0x0000020557F3EAD0>, body = None, file_name = '1663e52e-c26d-498b-a3ac-b4018c1ebd98-attachment.png'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = self._report_dir / file_name\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, str):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'NoneType'\n\n..\\..\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\logger.py:53: TypeError"}, "description": "Позитивный тест проверки удаления пользователем всех созданных им карточек питомцев. Валидация теста\n        выполнена успешно в случае, если после последовательного воздействия на элемент \"Удалить питомца\" в каждой\n        карточке питомца, все карточки будут удалены из стека пользователя. Тест предусматривает проверку количества\n        карточек до и после удаления.\n        P.S. По большей части тест создан для очистки профиля пользователя от ранее созданных карточек.", "steps": [{"name": "Шаг 1: Проверка количества карточек питомцев в профиле (если нет ни одной -> Exception).", "status": "passed", "start": 1705654167921, "stop": 1705654171290}, {"name": "Шаг 2: Удаление всех карточек питомцев из профиля", "status": "passed", "start": 1705654171290, "stop": 1705654274545}, {"name": "Шаг 3: Assert-проверка количества карточек до/после удаления.", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'\n", "trace": "  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject15_PageObjects_tests\\tests\\test_my_pets_positive.py\", line 197, in test_delete_all_pets_positive\n    allure.attach(screenshot,\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\_allure.py\", line 200, in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_callers.py\", line 113, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\allure_pytest\\listener.py\", line 249, in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\reporter.py\", line 165, in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_hooks.py\", line 493, in __call__\n    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_manager.py\", line 115, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_callers.py\", line 113, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\pluggy\\_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"C:\\Users\\agafo\\PycharmProjects\\pythonProject12\\venv\\lib\\site-packages\\allure_commons\\logger.py\", line 53, in report_attached_data\n    attached_file.write(body)\n"}, "attachments": [{"name": "delete_all_pets_PASSED", "source": "1663e52e-c26d-498b-a3ac-b4018c1ebd98-attachment.png", "type": "image/png"}], "start": 1705654274545, "stop": 1705654274647}], "start": 1705654167921, "stop": 1705654274648, "uuid": "b1cd1ae1-b980-4a29-a17d-f3894c8fff39", "historyId": "8f80541e0b85c42de40c217610312191", "testCaseId": "8f80541e0b85c42de40c217610312191", "fullName": "tests.test_my_pets_positive.TestMyPetsPagePositive#test_delete_all_pets_positive", "labels": [{"name": "severity", "value": "normal"}, {"name": "story", "value": "Удаление всех карточек питомцев из профиля пользователя (очистка профиля)"}, {"name": "feature", "value": "Удаление карточек питомцев_POSITIVE TESTS"}, {"name": "tag", "value": "delete_all_pets"}, {"name": "tag", "value": "four"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_my_pets_positive"}, {"name": "subSuite", "value": "TestMyPetsPagePositive"}, {"name": "host", "value": "DESKTOP-J32RPI2"}, {"name": "thread", "value": "13044-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_my_pets_positive"}]}